### ISSUE TITLE
spec3 crash in mir_mrs cube build with "not enough values to unpack"

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#8213]
_Issue [JP-3517](https://jira.stsci.edu/browse/JP-3517) was created on JIRA by [Hien Tran](https://jira.stsci.edu/secure/ViewProfile.jspa?name=htran):_

ops see a new failure in b10.0 for [jw01717-c1007_20240118t062156_spec3_00001](https://pljwdmsweb.stsci.edu/owlgui/logs/directory_view?directory_path=/ifs/archive/ops/jwst/info/owl/logs/owlmgr_jw01717-c1007_20240118t062156_spec3_00001_1705569182.879814&fileset_name=jw01717-c1007_20240118t062156_spec3_00001&dag_name=REPRO_LEVEL_3) during reprocessing. 
```
ValueError: not enough values to unpack (expected 15, got 13)
```
in cube_build step. this dataset made it successfully before in previous builds.
```
2024-01-18 09:16:46,269 - CRDS - DEBUG - Final effective context is 'jwst_1183.pmap'
2024-01-18 09:16:46,269 - CRDS - DEBUG - Computing best references locally.
2024-01-18 09:16:46,271 - CRDS - DEBUG - Bestrefs header:
{'META.EXPOSURE.READPATT [READPATT]': 'FASTR1',
'META.EXPOSURE.TYPE [EXP_TYPE]': 'MIR_MRS',
'META.INSTRUMENT.BAND [BAND]': 'LONG',
'META.INSTRUMENT.CHANNEL [CHANNEL]': '34.0',
'META.INSTRUMENT.CORONAGRAPH [CORONMSK]': 'UNDEFINED',
'META.INSTRUMENT.DETECTOR [DETECTOR]': 'MIRIFULONG',
'META.INSTRUMENT.FILTER [FILTER]': 'UNDEFINED',
'META.INSTRUMENT.LAMP_STATE [LAMP]': 'OFF',
'META.INSTRUMENT.NAME [INSTRUME]': 'MIRI',
'META.OBSERVATION.DATE [DATE-OBS]': '2023-06-10',
'META.OBSERVATION.TIME [TIME-OBS]': '06:42:27.176',
'META.SUBARRAY.NAME [SUBARRAY]': 'FULL',
'META.VISIT.TSOVISIT [TSOVISIT]': 'F',
'META.VISIT.TYPE [VISITYPE]': 'PRIME_TARGETED_FIXED',
'REFTYPE': 'UNDEFINED'}


...


2024-01-18 09:22:33,202 - stpipe.Spec3Pipeline.cube_build - INFO - Cube Geometry:
2024-01-18 09:22:33,202 - stpipe.Spec3Pipeline.cube_build - INFO - axis# Naxis CRPIX CRVAL CDELT(arcsec) Min & Max (xi, eta arcsec)
2024-01-18 09:22:33,202 - stpipe.Spec3Pipeline.cube_build - INFO - Axis 1 49 25.00 -153.82448096 0.13000000 -3.18499988 3.18499988
2024-01-18 09:22:33,203 - stpipe.Spec3Pipeline.cube_build - INFO - Axis 2 51 26.00 55.88707580 0.13000000 -3.31499988 3.31499988
2024-01-18 09:22:33,203 - stpipe.Spec3Pipeline.cube_build - INFO - axis# Naxis CRPIX CRVAL CDELT(microns) Min & Max (microns)
2024-01-18 09:22:33,203 - stpipe.Spec3Pipeline.cube_build - INFO - Axis 3 3438 1.00 4.90040010 0.00080000 4.90000010 7.65040003
2024-01-18 09:22:33,203 - stpipe.Spec3Pipeline.cube_build - INFO - Cube covers channel, subchannel: 1, short
2024-01-18 09:22:33,203 - stpipe.Spec3Pipeline.cube_build - INFO - Cube covers channel, subchannel: 1, medium
2024-01-18 09:22:33,203 - stpipe.Spec3Pipeline.cube_build - INFO - Cube covers channel, subchannel: 1, long
2024-01-18 09:22:33,203 - stpipe.Spec3Pipeline.cube_build - INFO - Subchannel listing: ['short', 'medium', 'long']
2024-01-18 09:22:33,203 - stpipe.Spec3Pipeline.cube_build - INFO - Output Name: jw01717-c1007_t001_miri_ch1-shortmediumlong
2024-01-18 09:22:35,838 - stpipe.Spec3Pipeline.cube_build - INFO - # of detector pixels not mapped to output plane: 3516 with wavelength below 4.898601598881583
Traceback (most recent call last):
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/bin/strun", line 26, in <module>
step = Step.from_cmdline(sys.argv[1:])
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/stpipe/step.py", line 186, in from_cmdline
return cmdline.step_from_cmdline(args)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/stpipe/cmdline.py", line 386, in step_from_cmdline
step.run(*positional)
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/stpipe/step.py", line 478, in run
step_result = self.process(*args)
^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/pipeline/calwebb_spec3.py", line 240, in process
result = self.cube_build(result)
^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/stpipe/step.py", line 478, in run
step_result = self.process(*args)
^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/cube_build/cube_build_step.py", line 375, in process
result, status = thiscube.build_ifucube()
^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/cube_build/ifu_cube.py", line 619, in build_ifucube
coord1, coord2, corner_coord, wave, dwave, flux, err, slice_no, rois_pixel, \
ValueError: not enough values to unpack (expected 15, got 13)
2024018092243 INFO src=ssb_calibration_wrapper._strun_analyze_and_log_failure fsn=jw01717-c1007_20240118t062156_spec3_00001_asn msg="strun /dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/pipeline/calwebb_spec3.cfg FAILED (exit=1) on jw01717-c1007_20240118t062156_spec3_00001_asn.json."
2024018092243 ERROR src=ssb_calibration_wrapper.calibrate._strun._strun_analyze_and_log_failure fsn=jw01717-c1007_20240118t062156_spec3_00001_asn msg="strun exit status=1" ```

**Comment by stscijgbot-jp** [comment]
_Comment by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) on [JIRA](https://jira.stsci.edu/browse/JP-3517?focusedCommentId=680551#comment-680551):_

Looks like this program is a little weird; Observation 1 has three bands of science observation, and Observation 2 was meant to have three bands of background observations.  However, the observations failed during the SHORT grating backgrounds, so there are no 1A/2A/3A/4A background data. 

I think that must be the root cause of this failure, as I can reproduce it offline using the B10.0.1 pipeline.  This issue should be fixed by #8001 which is in B10.1, as the crash does not repeat with that version of the code.

[Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) [Kirsten Larson](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kilarson) However, we may need to give attention to master background subtraction.  It looks like this is getting confused when trying to subtract a master background in Ch1A, without any idea what the Ch1A background actually is.  Looks like the missing background values are being interpreted as zero for most channels, but as NaN for Ch1A, likely because they're beyond the bounds of the input background spectrum.  What do we want the behavior to be?

**Comment by jemorrison** [comment]
I can not find this association on MAST. 
For jwst01717-c1007 I just have this association instead:
jw01717-c1007_20231106t154603_spec3_00001_asn
Did you remake the association or did I just miss finding it on MAST ?

**Comment by stscijgbot-jp** [comment]
_Comment by [Hien Tran](https://jira.stsci.edu/secure/ViewProfile.jspa?name=htran) on [JIRA](https://jira.stsci.edu/browse/JP-3517?focusedCommentId=680584#comment-680584):_

Jane, it's the same association. when we reprocess, it's renamed with a new timestamp. the input members are
```
jw01717001001_02101_00001_mirifulong_cal.fits science
jw01717001001_02101_00001_mirifushort_cal.fits science
jw01717001001_02101_00002_mirifulong_cal.fits science
jw01717001001_02101_00002_mirifushort_cal.fits science
jw01717001001_02101_00003_mirifulong_cal.fits science
jw01717001001_02101_00003_mirifushort_cal.fits science
jw01717001001_02101_00004_mirifulong_cal.fits science
jw01717001001_02101_00004_mirifushort_cal.fits science
jw01717001001_04101_00001_mirifulong_cal.fits science
jw01717001001_04101_00001_mirifushort_cal.fits science
jw01717001001_04101_00002_mirifulong_cal.fits science
jw01717001001_04101_00002_mirifushort_cal.fits science
jw01717001001_04101_00003_mirifulong_cal.fits science
jw01717001001_04101_00003_mirifushort_cal.fits science
jw01717001001_04101_00004_mirifulong_cal.fits science
jw01717001001_04101_00004_mirifushort_cal.fits science
jw01717001001_06101_00001_mirifulong_cal.fits science
jw01717001001_06101_00001_mirifushort_cal.fits science
jw01717001001_06101_00002_mirifulong_cal.fits science
jw01717001001_06101_00002_mirifushort_cal.fits science
jw01717001001_06101_00003_mirifulong_cal.fits science
jw01717001001_06101_00003_mirifushort_cal.fits science
jw01717001001_06101_00004_mirifulong_cal.fits science
jw01717001001_06101_00004_mirifushort_cal.fits science
jw01717002001_02101_00001_mirifulong_x1d.fits background
jw01717002001_02101_00001_mirifushort_x1d.fits background
jw01717002001_02101_00002_mirifulong_x1d.fits background
jw01717002001_02101_00002_mirifushort_x1d.fits background ```

**Comment by stscijgbot-jp** [comment]
_Comment by [Hien Tran](https://jira.stsci.edu/secure/ViewProfile.jspa?name=htran) on [JIRA](https://jira.stsci.edu/browse/JP-3517?focusedCommentId=680815#comment-680815):_

[Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) we just saw a bunch of failures for program 1246 with the same error message _"not enough values to unpack (expected 15, got 13)",_ but this time it's for spec2, not spec3. is it due to the same issue?
```
jw01246-o051_20240124t032321_spec2_00002 jw01246051001_03101_00002_mirifulong
jw01246-o051_20240124t032321_spec2_00004 jw01246051001_03101_00001_mirifulong
jw01246-o052_20240124t032321_spec2_00002 jw01246052001_03105_00004_mirifulong
jw01246-o052_20240124t032321_spec2_00004 jw01246052001_03105_00003_mirifulong
jw01246-o052_20240124t032321_spec2_00006 jw01246052001_03105_00002_mirifulong
jw01246-o052_20240124t032321_spec2_00008 jw01246052001_03105_00001_mirifulong
jw01246-o052_20240124t032321_spec2_00010 jw01246052001_03103_00004_mirifulong
jw01246-o052_20240124t032321_spec2_00016 jw01246052001_03103_00001_mirifulong
jw01246-o052_20240124t032321_spec2_00018 jw01246052001_03101_00004_mirifulong
jw01246-o052_20240124t032321_spec2_00014 jw01246052001_03103_00002_mirifulong
jw01246-o052_20240124t032321_spec2_00012 jw01246052001_03103_00003_mirifulong
jw01246-o052_20240124t032321_spec2_00020 jw01246052001_03101_00003_mirifulong
jw01246-o002_20240124t032321_spec2_00006 jw01246002001_03105_00002_mirifulong
jw01246-o052_20240124t032321_spec2_00022 jw01246052001_03101_00002_mirifulong
jw01246-o052_20240124t032321_spec2_00024 jw01246052001_03101_00001_mirifulong
jw01246-o002_20240124t032321_spec2_00010 jw01246002001_03103_00004_mirifulong
jw01246-o002_20240124t032321_spec2_00014 jw01246002001_03103_00002_mirifulong
jw01246-o002_20240124t032321_spec2_00012 jw01246002001_03103_00003_mirifulong
jw01246-o002_20240124t032321_spec2_00016 jw01246002001_03103_00001_mirifulong
jw01246-o002_20240124t032321_spec2_00002 jw01246002001_03105_00004_mirifulong
jw01246-o002_20240124t032321_spec2_00004 jw01246002001_03105_00003_mirifulong
jw01246-o002_20240124t032321_spec2_00008 jw01246002001_03105_00001_mirifulong
jw01246-o002_20240124t032321_spec2_00018 jw01246002001_03101_00004_mirifulong
jw01246-o002_20240124t032321_spec2_00020 jw01246002001_03101_00003_mirifulong
jw01246-o002_20240124t032321_spec2_00022 jw01246002001_03101_00002_mirifulong
jw01246-o053_20240124t032321_spec2_00004 jw01246053001_03105_00003_mirifulong
jw01246-o002_20240124t032321_spec2_00024 jw01246002001_03101_00001_mirifulong
jw01246-o053_20240124t032321_spec2_00002 jw01246053001_03105_00004_mirifulong
jw01246-o053_20240124t032321_spec2_00008 jw01246053001_03105_00001_mirifulong
jw01246-o053_20240124t032321_spec2_00006 jw01246053001_03105_00002_mirifulong
jw01246-o053_20240124t032321_spec2_00010 jw01246053001_03103_00004_mirifulong
jw01246-o053_20240124t032321_spec2_00012 jw01246053001_03103_00003_mirifulong
jw01246-o053_20240124t032321_spec2_00014 jw01246053001_03103_00002_mirifulong
jw01246-o053_20240124t032321_spec2_00018 jw01246053001_03101_00004_mirifulong
jw01246-o053_20240124t032321_spec2_00016 jw01246053001_03103_00001_mirifulong
jw01246-o053_20240124t032321_spec2_00022 jw01246053001_03101_00002_mirifulong
jw01246-o053_20240124t032321_spec2_00020 jw01246053001_03101_00003_mirifulong
jw01246-o053_20240124t032321_spec2_00024 jw01246053001_03101_00001_mirifulong
jw01246-o001_20240124t032321_spec2_00004 jw01246001001_03105_00003_mirifulong
jw01246-o001_20240124t032321_spec2_00002 jw01246001001_03105_00004_mirifulong
jw01246-o001_20240124t032321_spec2_00006 jw01246001001_03105_00002_mirifulong
jw01246-o001_20240124t032321_spec2_00008 jw01246001001_03105_00001_mirifulong
jw01246-o001_20240124t032321_spec2_00010 jw01246001001_03103_00004_mirifulong
jw01246-o001_20240124t032321_spec2_00012 jw01246001001_03103_00003_mirifulong
jw01246-o001_20240124t032321_spec2_00014 jw01246001001_03103_00002_mirifulong
jw01246-o001_20240124t032321_spec2_00016 jw01246001001_03103_00001_mirifulong
jw01246-o001_20240124t032321_spec2_00018 jw01246001001_03101_00004_mirifulong
jw01246-o001_20240124t032321_spec2_00020 jw01246001001_03101_00003_mirifulong
jw01246-o001_20240124t032321_spec2_00022 jw01246001001_03101_00002_mirifulong
jw01246-o001_20240124t032321_spec2_00024 jw01246001001_03101_00001_mirifulong ```

**Comment by jemorrison** [comment]
I will look into it. This type of error usually indicates that there is not valid data on the detector - so it signals another problem.

**Comment by jemorrison** [comment]
I doubled checked PID 1246 (Jupiter) and the data is saturated on many of the cal files. There are only 4 groups. The first frame correction- rejects the first frame. Looking at a few of the  uncal files, saturation often occurs on the second group - but now always. @drlaw1558 I don't know if some science could be obtained from such data by removing the first frame rejection. In such bright data the first frame is likely affected by the reset (especially in the second integration) but maybe something could be pulled out of the first integration. Something like this would be for off-line processing by the science team and not default processing by the pipeline.

**Comment by stscijgbot-jp** [comment]
_Comment by [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) on [JIRA](https://jira.stsci.edu/browse/JP-3517?focusedCommentId=681330#comment-681330):_

I am closing this ticket. The code runs in 10.1. In 10.0 there was a bug that when no valid data is found on the detector the code module giving the error should have returned 15 zero  length arrays, it returned 13 zero length arrays. This got fixed in 10.1
The underlying code issue is that there is no data on the detector. This happened in two different data sets 
1. The master background step did have all the data that was covered in the association (because the background observations failed). When this master background was subtracted it subtracted NANs for ch 1 A and for most of the pixels in ch 1 B. A ticket was filled to  fix (masterbackground) <https://jira.stsci.edu/browse/JP-3523.>
2. The input data was saturated and therefore no valid rate value is produced by the calwebb_detector 1 pipeline.

**Comment by stscijgbot-jp** [comment]
_Comment by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) on [JIRA](https://jira.stsci.edu/browse/JP-3517?focusedCommentId=681410#comment-681410):_

Sounds good, thanks for investigating.  Indeed, a while ago now I wrote a super-hacky notebook to try getting out rate files from 1-group data so that this team could move forward.  No idea if they're using my notebook still or not, but agreed that this is beyond default pipeline capability just now.
